cli/pcluster/utils.py

```diff
@@ -580,3 +598,13 @@ def retry(func, func_args, attempts=1, wait=0):
 
             LOGGER.debug("{0}, retrying in {1} seconds..".format(e, wait))
             time.sleep(wait)
+
+
+def retry_on_boto3_throttling(func, wait=5, *args, **kwargs):
+    while True:
+        try:
+            return func(*args, **kwargs)
+        except ClientError as e:
+            if e.response["Error"]["Code"] != "Throttling":
+                raise
+            time.sleep(wait)
```

Comment:
I'd add a warning message in the log file (or at least a debug message).
In this way if the CLI hangs we are able to understand the cause is the Throttling by checking the CLI logs.